Include/internal/pycore_frame.h

```diff
@@ -76,14 +80,27 @@ typedef struct _PyInterpreterFrame {
 } _PyInterpreterFrame;
 
 #define _PyInterpreterFrame_LASTI(IF) \
-    ((int)((IF)->instr_ptr - _PyCode_CODE(_PyFrame_GetCode(IF))))
+    ((int)((IF)->instr_ptr - _PyFrame_GetBytecode((IF))))
 
 static inline PyCodeObject *_PyFrame_GetCode(_PyInterpreterFrame *f) {
     PyObject *executable = PyStackRef_AsPyObjectBorrow(f->f_executable);
     assert(PyCode_Check(executable));
     return (PyCodeObject *)executable;
 }
 
+static inline _Py_CODEUNIT *
+_PyFrame_GetBytecode(_PyInterpreterFrame *f)
```

Comment:
> You were storing the bytecode in the frame directly before, IIRC.

Yep. You suggested storing `tlbc_index` instead since it was smaller. I think this is better for a couple of reasons:

1. It's smaller, as you said.
2. It simplifies the implementation of `RESUME_CHECK`. Previously, we would have to load the bytecode pointer for the current thread and deopt if it didn't match what was in the frame. Now we only have to compare tlbc indices. This is a cost shift, however, since now the callers of `_PyFrame_GetBytecode` have to do the more expensive load of the bytecode. I think the size reduction + simplification of `RESUME_CHECK` probably outweighs the higher cost of `_PyFrame_GetBytecode`.